docs/resources/server/damagetypes.md

```diff
@@ -0,0 +1,127 @@
+# Damage Types & Damage Sources
+
+A damage type denotes what kind of damage is being applied to an entity - physical damage, fire damage, drowning damage, magic damage, void damage, etc. The distinction into damage types is used for various immunities (e.g. blazes won't take fire damage), enchantments (e.g. blast protection will only protect against explosion damage), and many more use cases.
+
+A damage type is a template for a damage source, so to speak. Or in other words, a damage source can be viewed as a damage type instance. Damage types are registered in code, but have all of their properties defined in data packs. Damage sources, on the other hand, are created as needed by the game, based off the values in the data pack files. They can hold additional context, for example the attacking entity.
+
+## Registering Damage Types
+
+To get started, you want to create your own `DamageType`. `DamageType`s are a [datapack registry][dr], and as such, new `DamageType`s are not registered in code, but are registered automatically when the corresponding files are added. However, we still need to provide some point for the code to get the damage sources from. We do so by specifying a [resource key][rk]:
+
+```java
+public static final ResourceKey<DamageType> EXAMPLE_DAMAGE =
+        ResourceKey.create(Registries.DAMAGE_TYPE, new ResourceLocation(ExampleMod.MOD_ID, "example"));
+```
+
+Now that we can reference it from code, let's specify some properties in the data file. Our data file is located at `data/examplemod/damage_type/example.json` (swap out `examplemod` and `example` for the mod id and the name of the resource location) and contains the following:
+
+```json5
+{
+  // The death message id of the damage type. The full death message translation key will be
+  // "death.examplemod.example" (with swapped-out mod ids and names).
+  "message_id": "example",
+  // Whether this damage type's damage amount scales with difficulty or not. Valid values are:
+  // - "never": The damage value remains the same on any difficulty. Common for player-caused damage types.
+  // - "when_caused_by_living_non_player": The damage value is scaled if the entity is caused by a
+  //   living entity of some sort, including indirectly (e.g. an arrow shot by a skeleton), that is not a player.
+  // - "always": The damage value is always scaled. Commonly used by explosion-like damage.
+  "scaling": "when_caused_by_living_non_player",
+  // The amount of exhaustion caused by receiving this kind of damage.
+  "exhaustion": 0.1,
+  // The damage effects (currently only sound effects) that are applied when receiving this kind of damage. Optional.
+  // Valid values are "hurt" (default), "thorns", "drowning", "burning", "poking", and "freezing".
+  "effects": "hurt",
+  // The death message type. Determines how the death message is built. Optional.
+  // Valid values are "default" (default), "fall_variants", and "intentional_game_design".
+  "death_message_type": "default"
```

Comment:
Valid 'vanilla' values yes. NeoForge provides extensible enums, so this can be added to depending on the modder. I know you say it below, but assume people are going to take your word for it.